internal/output/templates.go

```diff
@@ -157,14 +166,76 @@ iVBORw0KGgoAAAANSUhEUgAAAMAAAADACAMAAABlApw1AAAABGdBTUEAALGPC/xhBQAAAAFzUkdCAK7O
       {{if gt .Indent 0}}<span class="arrow">&#8627;</span>{{end}}
       {{.CostComponent.Name}}
     </td>
-    <td class="monthly-quantity">{{.CostComponent.MonthlyQuantity | formatQuantity }}</td>
-    <td class="unit">{{.CostComponent.Unit}}</td>
-    <td class="price">{{.CostComponent.Price | formatPrice }}</td>
-    <td class="hourly-cost">{{.CostComponent.HourlyCost | formatCost2DP}}</td>
-    <td class="monthly-cost">{{.CostComponent.MonthlyCost | formatCost2DP}}</td>
+    {{if .CostComponent.MonthlyCost}}
+      {{if contains .Fields "monthlyQuantity"}}
+        <td class="monthly-quantity">{{.CostComponent.MonthlyQuantity | formatQuantity }}</td>
+      {{end}}
+      {{if contains .Fields "unit"}}
+        <td class="unit">{{.CostComponent.Unit}}</td>
+      {{end}}
+      {{if contains .Fields "price"}}
+        <td class="price">{{.CostComponent.Price | formatPrice }}</td>
+      {{end}}
+      {{if contains .Fields "hourlyCost"}}
+        <td class="hourly-cost">{{.CostComponent.HourlyCost | formatCost2DP}}</td>
+      {{end}}
+      {{if contains .Fields "monthlyCost"}}
+        <td class="monthly-cost">{{.CostComponent.MonthlyCost | formatCost2DP}}</td>
+      {{end}}
+    {{else}}
+      <td colspan="{{len .Fields}}" class="usage-cost">Cost depends on usage: {{.CostComponent.Price | formatPrice}} per {{.CostComponent.Unit}}</td>
```

Comment:
Should we make the text color `#6b7280` here so it's a little faded like in the CLI?